src/ast/ast_validations.c

```diff
@@ -1592,29 +1592,35 @@ bool AST_ContainsErrors(const cypher_parse_result_t *result) {
 	return cypher_parse_result_nerrors(result) > 0;
 }
 
-static AST_Validation _AST_Validate_ParseResultRoot(const cypher_parse_result_t *result) {
+static AST_Validation _AST_Validate_ParseResultRoot(const cypher_parse_result_t *result,
+													int *index) {
 	// Check for failures in libcypher-parser
 	if(AST_ContainsErrors(result)) {
 		_AST_ReportErrors(result);
 		return AST_INVALID;
 	}
 
-	const cypher_astnode_t *root = cypher_parse_result_get_root(result, 0);
-	// Check for empty query
-	if(root == NULL) {
-		QueryCtx_SetError("Error: empty query.");
-		return AST_INVALID;
-	}
-
-	cypher_astnode_type_t root_type = cypher_astnode_type(root);
-	if(root_type != CYPHER_AST_STATEMENT) {
-		// This should be unnecessary, as we're currently parsing
-		// with the CYPHER_PARSE_ONLY_STATEMENTS flag.
-		QueryCtx_SetError("Encountered unsupported query type '%s'", cypher_astnode_typestr(root_type));
-		return AST_INVALID;
+	uint nroots = cypher_parse_result_nroots(result);
+	for(uint i = 0; i < nroots; i++) {
+		const cypher_astnode_t *root = cypher_parse_result_get_root(result, i);
+		cypher_astnode_type_t root_type = cypher_astnode_type(root);
+		if(root_type == CYPHER_AST_LINE_COMMENT || root_type == CYPHER_AST_BLOCK_COMMENT ||
+		   root_type == CYPHER_AST_COMMENT) {
+			continue;
+		} else if(root_type != CYPHER_AST_STATEMENT) {
+			// This should be unnecessary, as we're currently parsing
+			// with the CYPHER_PARSE_ONLY_STATEMENTS flag.
```

Comment:
I think this comment should be deleted, as we might get here while parsing parameters if I understand right.